proto/p2p/p2p.proto

```diff
@@ -58,6 +58,7 @@ message Message {
     AppGossip app_gossip = 32;
 
     PeerListAck peer_list_ack = 33;
+    AppRequestFailed app_request_failed = 34;
```

Comment:
> We should probably consider grouping these with their related messages... I do like having the numbers in order... (as it is easy to verify which number to use next)...

Yeah I originally went with this approach for that reason since I saw it as a trade-off between pretty grouping versus it being annoying to figure out which number we need to use.

> But I've also seen people explicitly reserveing the next number to use as an alternative way of tracking which numbers have been used

I'm not following here... how does this help group the related messages together?

Reply:
> how does this help group the related messages together

It lets us group related messages together while also being very explicit about the next number to use